nncf/experimental/common/quantization/algorithms/quantizer/fx_quantizer.py

```diff
@@ -0,0 +1,130 @@
+# Copyright (c) 2024 Intel Corporation
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#      http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+from collections import defaultdict
+from copy import deepcopy
+from typing import Dict, Tuple, Union
+
+import torch
+import torch.fx
+from torch.ao.quantization.quantizer import Quantizer
+from torch.ao.quantization.quantizer.quantizer import QuantizationSpec
+from torch.ao.quantization.quantizer.quantizer import QuantizationSpecBase
+from torch.ao.quantization.quantizer.quantizer import SharedQuantizationSpec
+
+import nncf
+from nncf.common.graph.graph import NNCFGraph
+from nncf.common.quantization.quantizer_setup import ActivationQuantizationInsertionPoint
+from nncf.common.quantization.quantizer_setup import SingleConfigQuantizationPoint
+from nncf.common.quantization.quantizer_setup import SingleConfigQuantizerSetup
+from nncf.common.quantization.quantizer_setup import WeightQuantizationInsertionPoint
+from nncf.common.quantization.structs import QuantizationScheme as QuantizationMode
+from nncf.common.quantization.structs import QuantizerConfig
+from nncf.experimental.common.quantization.algorithms.quantizer.base_quantizer import NNCFQuantizer
+
+EdgeOrNode = Union[Tuple[torch.fx.Node, torch.fx.Node]]
+
+
+class NNCFFXQuantizer(NNCFQuantizer):
```

Comment:
Take into account the OpenVINO Quantizer implementation https://github.com/openvinotoolkit/nncf/blob/44ebb6cdb78a52c7862a6a9e27daad45c343de3a/nncf/experimental/common/quantization/algorithms/quantizer/openvino_quantizer.py#L45, I would suggest to make some refactorng:
1. Does not inhered OpenVINO Qunatizer from `nncf.Quantizer` for simplification up-streaming to PyTorch.
2. Introduce adapters for torch.ao Quantizers and OpenVINO Quantizer, to avoid repacking quantization setup: `TorchAOQuantizerAdapter`, `OpenVINOQuantizerAdapter`. Declaration can be the following:
```
class TorchAOQuantizerAdapter(nncf.Quantizer, torch.ao.Quantizer)
```
```
class OpenVINOQuantizerAdapter(TorchAOQuantizerAdapter)
```

Reply:
`OpenVINOQuantizer` will be introduced in the following PR, I'll apply this to the following PR

Reply:
I believe, that the part of this comment can be applied in this PR. I'm open to discuss it offline.